src/rbd.cc

```diff
@@ -3034,6 +3065,30 @@ if (!set_conf_param(v, p1, p2, p3)) { \
   // the relevant parts
   set_pool_image_name(imgname, (char **)&poolname,
 		      (char **)&imgname, (char **)&snapname);
+
+
+  if(!is_valid_name(poolname)) 
+    invalid_count++;
+  
+  if(!is_valid_name(snapname)) 
+    invalid_count++;
+
+  if(!is_valid_name(imgname)) 
+    invalid_count++;
+  
+
+  if(invalid_count > 0) {
```

Comment:
There must be a better way to do this, so you show exactly which items are invalid.  Also, this whole section should be in a function since almost the same code is used.  Pass a string to insert either "" or "destination " in the output depending on if it is called from here or below, respectively.

Maybe something like this:

int check_names(string dest, string pool, string snap, string image)
{
char *names [3];
if invalid pool name
   names[invalid_count++] = "pool"
if invalid snap name
   names[invalid_count++] = "snap"
if invalid image name
   names[invalid_count++] = "image"

if invalid_count == 0
    return 0;

 ostringstream items;
 items << names[0];
 if (invalid_count == 1) {
     plural = " is "
  } else {
     plural = "s are"
    if (invalid_count == 2)
      items << " and " << names[1]
    else
      items << ", " << names[1] << " and " << names[2];
  }
  cerr << "rbd: The " << dest << items.str() << " name" << plural << " invalid" << std::endl;
  return EXIT_FAILURE;
}

or just keep it super simple and output a message for each item so you might see.

rbd: The pool name is invalid
rbd: The image name is invalid
rbd: The snap name is invalid